File: check_columns.js
const axios = require('axios');
const xlsx = require('xlsx');
const fs = require('fs');

async function analyze() {
    const doc_id = '1vWHXdrgN6JUstWk6g5cXuLJwYJRkniObUcFgBMs4108';
    const url = `https://docs.google.com/spreadsheets/d/${doc_id}/export?format=xlsx`;
    const results = {};
    
    try {
        console.log('Fetching spreadsheet...');
        const response = await axios.get(url, { responseType: 'arraybuffer' });
        const workbook = xlsx.read(response.data, { type: 'buffer' });
        const sheetName = 'Semana';
        
        if (workbook.SheetNames.includes(sheetName)) {
            const sheet = workbook.Sheets[sheetName];
            const data = xlsx.utils.sheet_to_json(sheet, { header: 1 });
            const columns = data[0];
            results.columns = columns;
            
            const targetCols = ['Receita / Visitas', 'Order', 'Visitas', 'Revenue'];
            results.analysis = {};
            targetCols.forEach(tc => {
                const matches = columns.filter(col => col && col.toString().toLowerCase().includes(tc.toLowerCase()));
                results.analysis[tc] = {
                    found: matches.length > 0,
                    matches: matches
                };
            });
            
            results.sample = xlsx.utils.sheet_to_json(sheet).slice(0, 2);
        } else {
            results.error = `Sheet '${sheetName}' not found. Available: ${workbook.SheetNames}`;
        }
    } catch (e) {
        results.error = e.message;
    }
    
    fs.writeFileSync('column_analysis.json', JSON.stringify(results, null, 2));
    console.log('Done.');
}

analyze();

File: check_columns.py
import pandas as pd
import urllib.request
import io
import json

def analyze_columns(doc_id, sheet_name):
    url = f"https://docs.google.com/spreadsheets/d/{doc_id}/export?format=xlsx"
    results = {}
    try:
        req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
        with urllib.request.urlopen(req) as response:
            file_content = response.read()
            xls = pd.ExcelFile(io.BytesIO(file_content))
            
            if sheet_name in xls.sheet_names:
                df = pd.read_excel(io.BytesIO(file_content), sheet_name=sheet_name)
                cols = df.columns.tolist()
                results['columns'] = cols
                
                target_cols = ['Receita / Visitas', 'Order', 'Visitas', 'Revenue']
                results['analysis'] = {}
                for tc in target_cols:
                    matches = [col for col in cols if tc.lower() in str(col).lower()]
                    results['analysis'][tc] = {
                        'found': len(matches) > 0,
                        'matches': matches
                    }
                
                results['sample'] = df.head(2).to_dict(orient='records')
            else:
                results['error'] = f"Sheet '{sheet_name}' not found"

    except Exception as e:
        results['error'] = str(e)
    
    with open('column_analysis.json', 'w', encoding='utf-8') as f:
        json.dump(results, f, indent=2, ensure_ascii=False)

analyze_columns('1vWHXdrgN6JUstWk6g5cXuLJwYJRkniObUcFgBMs4108', 'Semana')
